src/include/sof/audio/sink_api_implementation.h

```diff
@@ -11,72 +11,7 @@
 #include <stdint.h>
 #include <stdbool.h>
 
-/* forward def */
-struct sof_audio_stream_params;
```

Comment:
As we agreed some times ago, we'll give up with total separation of sink/src internals and externals (like private/public in c++). 
All inlines should go to sinlk_api.h / source_api.h
__api_implementation.h  is no longer needed